diesel/src/expression/expression_methods/sort_expression_methods.rs

```diff
@@ -0,0 +1,65 @@
+use expression::predicates::*;
+
+#[cfg(feature = "postgres")]
+pub trait Ordering : Sized {
+    /// Specify that nulls should come before other values in this ordering.
+    /// Normally, nulls come last when sorting in ascending order and first
+    /// when sorting in descending order.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// let query = users.select(id).order(name.asc().nulls_first());
+    /// assert_eq!("SELECT `users`.`id` FROM `users` ORDER BY `users`.`name` ASC NULLS FIRST",
+    ///            debug_sql!(query));
```

Comment:
I'd prefer to avoid testing the generated SQL, and instead demonstrate the results in a data set.